### ISSUE TITLE
chaos-mesh CPU and Memory stress doesn't work on k8s 1.25 and containerd

### ISSUE THREAD

**Issue by robbie1douglas** [chaos-mesh/chaos-mesh#4038]
## Bug Report
Could this be related to CGroup v2 on containerd support? It looks like CGroup v2 is supported on Docker and CRI-O but I can't find reference for it being supported on containerd yet. Ref: https://github.com/chaos-mesh/chaos-mesh/issues/3559 and https://github.com/chaos-mesh/chaos-mesh/issues/2069

**What version of Kubernetes are you using?**
1.25

**What version of Chaos Mesh are you using?**
2.5.1

**What did you do? / Minimal Reproducible Example**
`apiVersion: chaos-mesh.org/v1alpha1
kind: StressChaos
metadata:
  name: cpu-stress
  namespace: chaos-mesh-test
spec:
  mode: all
  containerNames: ['sample-app']
  selector:
    labelSelectors:
      'app.kubernetes.io/name': 'sample-app'
  stressors:
    cpu:
      workers: 1
      load: 50 `

**What did you expect to see?**
Stress on pod

**What did you see instead?**
`k describe stresschaos.chaos-mesh.org/cpu-stress -n chaos-mesh-test
Name:         cpu-stress
Namespace:    chaos-mesh-test
Labels:       <none>
Annotations:  <none>
API Version:  chaos-mesh.org/v1alpha1
Kind:         StressChaos
Metadata:
  Creation Timestamp:  2023-05-09T01:07:41Z
  Finalizers:
    chaos-mesh/records
  Generation:  19
  Managed Fields:
    API Version:  chaos-mesh.org/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:containerNames:
        f:mode:
        f:selector:
          .:
          f:labelSelectors:
            .:
            f:app.kubernetes.io/name:
        f:stressors:
          .:
          f:cpu:
            .:
            f:load:
            f:workers:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2023-05-09T01:07:41Z
    API Version:  chaos-mesh.org/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .:
          v:"chaos-mesh/records":
      f:status:
        f:conditions:
        f:experiment:
          f:containerRecords:
          f:desiredPhase:
    Manager:         chaos-controller-manager
    Operation:       Update
    Time:            2023-05-09T01:07:55Z
  Resource Version:  307535670
  UID:               7b1aeb43-caf6-47e3-b14c-9b1a7c0d868a
Spec:
  Container Names:
    sample-app
  Mode:  all
  Selector:
    Label Selectors:
      app.kubernetes.io/name:  sample-app
    Namespaces:
      chaos-mesh-test
  Stressors:
    Cpu:
      Load:     50
      Workers:  1
Status:
  Conditions:
    Status:  True
    Type:    Selected
    Status:  False
    Type:    AllInjected
    Status:  True
    Type:    AllRecovered
    Status:  False
    Type:    Paused
  Experiment:
    Container Records:
      Events:
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:42Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:42Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:43Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:43Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:45Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:45Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:55Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:55Z
        Type:           Failed
      Id:               chaos-mesh-test/rdtest-sample-app-759c5b4546-xswd4/sample-app
      Injected Count:   0
      Phase:            Not Injected
      Recovered Count:  0
      Selector Key:     .
      Events:
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:41Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:42Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:42Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:43Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:43Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:45Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:45Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:55Z
        Type:           Failed
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
        Operation:      Apply
        Timestamp:      2023-05-09T01:07:55Z
        Type:           Failed
      Id:               chaos-mesh-test/rdtest-sample-app-759c5b4546-b54td/sample-app
      Injected Count:   0
      Phase:            Not Injected
      Recovered Count:  0
      Selector Key:     .
    Desired Phase:      Run
Events:
  Type     Reason           Age   From            Message
  ----     ------           ----  ----            -------
  Normal   FinalizerInited  30s   initFinalizers  Finalizer has been inited
  Normal   Updated          30s   initFinalizers  Successfully update finalizer of resource
  Normal   Started          30s   desiredphase    Experiment has started
  Normal   Updated          30s   desiredphase    Successfully update desiredPhase of resource
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          30s   records         Successfully update records of resource
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          30s   records         Successfully update records of resource
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          30s   records         Successfully update records of resource
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          30s   records         Successfully update records of resource
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          30s   records         Successfully update records of resource
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          30s   records         Successfully update records of resource
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           30s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          30s   records         Successfully update records of resource
  Warning  Failed           29s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           29s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          29s   records         Successfully update records of resource
  Warning  Failed           29s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           29s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          29s   records         Successfully update records of resource
  Warning  Failed           28s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           28s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          28s   records         Successfully update records of resource
  Warning  Failed           28s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           28s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          28s   records         Successfully update records of resource
  Warning  Failed           26s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           26s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          26s   records         Successfully update records of resource
  Warning  Failed           26s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           26s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          26s   records         Successfully update records of resource
  Warning  Failed           16s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           16s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          16s   records         Successfully update records of resource
  Warning  Failed           16s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226771: exit status 1
  Warning  Failed           16s   records         Failed to apply chaos: rpc error: code = Unknown desc = get cgroup path of pid 226897: exit status 1
  Normal   Updated          15s   records         Successfully update records of resource`
**Output of chaosctl**
Not available in my env.

**Comment by chrisjroy** [comment]
We are experiencing this issue as well in our AKS clusters running k8s 1.25 with containerd.  From our testing, the issue only applies to k8s 1.25 using containerd **running under cgroup v2**.  Once we applied this yaml (https://github.com/Azure/AKS/blob/master/examples/cgroups/revert-cgroup-v1.yaml) to revert our k8s 1.25 containerd cluster to cgroup v1, chaosmesh was able to successfully run cpu and memory attacks.

**Comment by aido123** [comment]
+1

Also seeing the same issue in AKS 1.25.

@STRRL, support was just added for cri-o and docker previously only?

**Comment by dstoecklein** [comment]
+ 1
Same issue in AKS 1.25 w/ containerd

**Comment by STRRL** [comment]
Hi folks! Sorry for the late response!

I would try to reproduce it on minikube with VM driver and contianerd CRI

**Comment by STRRL** [comment]
Hi folks!

tldr; I need more information to profile this issue. :heart: I would appreciate it very much if somebody could give us more detailed information like 

- Linux Distros
- Linux Kernel Version
- Related logs in chaos-daemon-xxx pod
- any other valueable infos

cc @robbie1douglas @chrisjroy @aido123 @dstoecklein 

I could NOT reproduce this issue on my local machine, with minikube docker driver, contained runtime, and cgroup v2:

Here is my clueter info:

```text
❯ k get nodes -o wide
NAME       STATUS   ROLES           AGE     VERSION    INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION   CONTAINER-RUNTIME
minikube   Ready    control-plane   4m51s   v1.25.11   192.168.49.2   <none>        Ubuntu 20.04.5 LTS   6.3.9-arch1-1    containerd://1.6.20

❯ helm -n chaos-mesh list   
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
chaos-mesh      chaos-mesh      2               2023-06-30 10:34:40.957905011 +0800 CST deployed        chaos-mesh-2.5.1        2.5.1   

```

I am sure that I have enabled the cgroupv2
```text
❯ minikube ssh
Last login: Fri Jun 30 02:37:09 2023 from 192.168.49.1
docker@minikube:~$ mount | grep cgroup
cgroup on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
```

Here is how I create the workload, and StressChaos spec:


```text
k create deployment nginx --replicas 1 --image nginx:latest
```

```yaml
apiVersion: chaos-mesh.org/v1alpha1
kind: StressChaos
metadata:
  name: burn-cpu
spec:
  mode: one
  selector:
    labelSelectors:
      app: nginx
  stressors:
    cpu:
      workers: 1
      load: 50
```

and StressChaos works well, stress appears on `top`, and no error log in chaos-daemon-xxx pod:


<img width="668" alt="image" src="https://github.com/chaos-mesh/chaos-mesh/assets/20221408/4deec2e0-4c42-49cb-9bef-28b028ee4c7a">


```text
❯ k -n chaos-mesh logs -f chaos-daemon-bcfxw 
Chaos-daemon Version: version.Info{GitVersion:"chart-2.5.1", GitCommit:"08b173528f315d3a6621b9be5566e74ab9e0d35d", BuildDate:"2023-01-14T02:33:17Z", GoVersion:"go1.18", Compiler:"gc", Platform:"linux/amd64"}
2023-06-30T02:35:24.347Z        INFO    chaos-daemon.daemon-server      chaos-daemon/main.go:80 grant access to /dev/fuse
2023-06-30T02:35:24.347Z        ERROR   chaos-daemon.daemon-server      chaos-daemon/main.go:83 grant access to /dev/fuse       {"error": "fail to find device cgroup", "errorVerbose": "fail to find device cgroup\ngithub.com/chaos-mesh/chaos-mesh/pkg/fusedev.GrantAccess\n\t/home/runner/work/chaos-mesh/chaos-mesh/pkg/fusedev/fusedev_linux.go:60\nmain.main\n\t/home/runner/work/chaos-mesh/chaos-mesh/cmd/chaos-daemon/main.go:81\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1571"}
main.main
        /home/runner/work/chaos-mesh/chaos-mesh/cmd/chaos-daemon/main.go:83
runtime.main
        /usr/local/go/src/runtime/proc.go:250
2023-06-30T02:35:24.347Z        INFO    chaos-daemon.daemon-server      chaos-daemon/main.go:86 remount /sys with read-write permission
2023-06-30T02:35:24.348Z        INFO    chaos-daemon.daemon-server      runtime/asm_amd64.s:1571        Starting grpc endpoint  {"address": "0.0.0.0:31767", "runtime": "containerd"}
2023-06-30T02:35:24.348Z        INFO    chaos-daemon.daemon-server      runtime/asm_amd64.s:1571        Starting http endpoint  {"address": "0.0.0.0:31766"}
2023-06-30T02:37:00.258Z        INFO    chaos-daemon.daemon-server      chaosdaemon/server.go:187       Executing stressors     {"namespacedName": "default/burn-cpu", "request": "target:\"containerd://1968ffa5945c4f0a2d0a7034a3ec44a4ed841a114de15667eefe82f64946217d\" cpuStressors:\" --cpu-load-slice 10 --cpu-method sqrt --cpu 1 --cpu-load 50\" enterNS:true"}
2023-06-30T02:37:00.263Z        INFO    chaos-daemon.daemon-server.background-process-manager.process-builder   chaosdaemon/stress_server_linux.go:39   build command   {"namespacedName": "default/burn-cpu", "command": "/usr/local/bin/pause /usr/local/bin/nsexec -p /proc/3108/ns/pid -- stress-ng --cpu-load-slice 10 --cpu-method sqrt --cpu 1 --cpu-load 50"}
2023-06-30T02:37:00.265Z        INFO    chaos-daemon.daemon-server      pb/chaosdaemon.pb.go:4322       Start stress-ng successfully    {"namespacedName": "default/burn-cpu", "command": "/usr/local/bin/pause /usr/local/bin/nsexec -p /proc/3108/ns/pid -- stress-ng --cpu-load-slice 10 --cpu-method sqrt --cpu 1 --cpu-load 50", "pid": 5045, "uid": "50719cf0-ab7e-40f7-9f45-c968a1bc3693"}
2023-06-30T02:37:00.344Z        INFO    chaos-daemon.daemon-server      pb/chaosdaemon.pb.go:4322       send signal to resume process   {"namespacedName": "default/burn-cpu"}
2023-06-30T02:37:00.345Z        INFO    chaos-daemon.daemon-server      pb/chaosdaemon.pb.go:4322       pause has been resumed  {"namespacedName": "default/burn-cpu", "comm": "nsexec\n"}

```

**Comment by STRRL** [comment]
it might mount both cgroup v1 and v2 at the same time. it could be figured by `mount | grep cgroup`, and check there exists both cgourp and cgoup2 mountpoints.

**Comment by dstoecklein** [comment]
My organization upgraded to K8s 1.26.3 just a few days ago and the problem does not occur anymore and the experiment runs as expected 😊 

I have another AKS cluster with K8s 1.25.6 w/ containerd and cgroup v1 installed (i checked with ``stat -fc %T /sys/fs/cgroup/``) and the experiment also runs as expected. So at the moment i have no way to reproduce this error 🤔

**Comment by aido123** [comment]
@STRRL I believe the issue is due to us not running the chaos demon DS as privileged. We were using the set of capabilities. Once we set the DS to privileged, it started to work.

I'm guessing the nsenter command in the code needs to be run in privileged mode. Is there a capability that can allow this rather than running in privileged?

**Comment by STRRL** [comment]
> @STRRL I believe the issue is due to us not running the chaos demon DS as privileged. We were using the set of capabilities. Once we set the DS to privileged, it started to work.
> 
> I'm guessing the nsenter command in the code needs to be run in privileged mode. Is there a capability that can allow this rather than running in privileged?

🤔 Got that. Could you give us the capability list as a reference?

**Comment by dekhtyarev** [comment]
Hi!

We have the same error.
When using capabilities - it doesn't work.
In privileged mode everything works.

Errors running tests:
```
$ kubectl -n devops-chaos describe stresschaos test-cpu-x113-failed                                                                
Name:         test-cpu-x113-failed
Namespace:    devops-chaos                                                                                                                                    
Labels:       <none>         
Annotations:  <none>                        
API Version:  chaos-mesh.org/v1alpha1
Kind:         StressChaos                                                                                                                                     
Metadata:                    
  Creation Timestamp:  2024-04-10T06:16:43Z 
  Finalizers:                 
    chaos-mesh/records                                                                                                                                        
  Generation:        19      
  Resource Version:  7264117728             
  UID:               b54f8a07-f5e8-470b-a0aa-8d5e171df412
Spec:                                                                                                                                                         
  Duration:  2m              
  Mode:      fixed                          
  Selector:                   
    Label Selectors:                                                                                                                                          
      App:  nginx            
    Namespaces:                             
      devops-chaos            
  Stressors:                                                            
    Cpu:                 
      Load:     50                  
      Workers:  1        
  Value:        1        
Status:                    
  Conditions:
    Status:  True                                        
    Type:    Selected                                    
    Status:  False                                                         
    Type:    AllInjected                                                                                                                                      
    Status:  True                                                       
    Type:    AllRecovered                                                                                                                                     
    Status:  False                                                                                                                                            
    Type:    Paused                                                                                                                                           
  Experiment:                                                                                                                                                 
    Container Records:                                                                                                                                        
      Events:                                                                                                                                                 
        Message:        rpc error: code = Unknown desc = get cgroup path of pid 92464: exit status 1
        Operation:      Apply                                                                                                                                 
        Timestamp:      2024-04-10T06:16:43Z                                                                                                                  
        Type:           Failed
...
```

Capabilities used
```
securityContext:
 capabilities:
   add:
   - SYS_PTRACE
   - NET_ADMIN
   - NET_RAW
   - MKNOD
   - SYS_CHROOT
   - SYS_ADMIN
   - KILL
   - IPC_LOCK
```

Mounted volumes:
```
spec:
   containers:
   - command:
      - /usr/local/bin/chaos-daemon
      - --runtime
      - containerd
      - --runtime-socket-path
      - /host-run/containerd.sock
      ...
      name: chaos-daemon
      ...
      volumeMounts:
      - mountPath: /host-run
         name: socket-path
      - mountPath: /host-sys
         name: sys-path
      - mountPath: /lib/modules
         name: lib-modules
      - mountPath: /etc/chaos-daemon/cert
         name: chaos-daemon-cert
         readOnly: true
   ...
   volumes:
   - hostPath:
         path: /run/containerd
         type: ""
      name: socket-path
   - hostPath:
         path: /sys
         type: ""
      name: sys-path
   - hostPath:
         path: /lib/modules
         type: ""
```

There are errors when remounting a directory in the daemon POD:
$ kubectl -n chaos-mesh logs chaos-daemon-zbbbk 
Chaos-daemon Version: version.Info{GitVersion:"chart-2.6.3", GitCommit:"b159d1c5a361ca350ca7789c9eb7fc917479db39", BuildDate:"2024-01-23T15:04:55Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}

2024-04-10T13:16:06.255+0700    INFO    chaos-daemon.daemon-server      chaos-daemon/main.go:80 grant access to /dev/fuse
2024-04-10T13:16:06.256+0700    ERROR   chaos-daemon.daemon-server      chaos-daemon/main.go:83 grant access to /dev/fuse       {"error": "fail to find device cgroup", "errorVerbose": "fail to find device cgroup\ngithub.com/chaos-mesh/chaos-mesh/pkg/fusedev.GrantAccess\n\t/home/runner/work/chaos-mesh/chaos-mesh/pkg/fusedev/fusedev_linux.go:60\nmain.main\n\t/home/runner/work/chaos-mesh/chaos-mesh/cmd/chaos-daemon/main.go:81\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1594"}
main.main
        /home/runner/work/chaos-mesh/chaos-mesh/cmd/chaos-daemon/main.go:83
runtime.main
        /usr/local/go/src/runtime/proc.go:250

2024-04-10T13:16:06.256+0700    INFO    chaos-daemon.daemon-server      chaos-daemon/main.go:86 remount /sys with read-write permission
2024-04-10T13:16:06.256+0700    ERROR   chaos-daemon.daemon-server      chaos-daemon/main.go:89 remount /sys with read-write permission {"error": "permission denied"}
main.main
        /home/runner/work/chaos-mesh/chaos-mesh/cmd/chaos-daemon/main.go:89
runtime.main
        /usr/local/go/src/runtime/proc.go:250

2024-04-10T13:16:06.259+0700    INFO    chaos-daemon.daemon-server      runtime/asm_amd64.s:1594        Starting http endpoint  {"address": "0.0.0.0:31766"}
2024-04-10T13:16:06.259+0700    INFO    chaos-daemon.daemon-server      runtime/asm_amd64.s:1594        Starting grpc endpoint  {"address": "0.0.0.0:31767", "runtime": "containerd"}
```

Mounting directories looks like this:
```
$ kubectl -n chaos-mesh exec -t chaos-daemon-zbbbk -- mount | grep cgroup
cgroup on /sys/fs/cgroup type cgroup2 (ro,nosuid,nodev,noexec,relatime)
cgroup on /host-run/io.containerd.runtime.v2.task/k8s.io/06ebffbec2c4c25b9276cfeebf322a2385040bb710eb4000cc88ae45487132b8/rootfs/sys/fs/cgroup type cgroup2 (ro,nosuid,nodev,noexec,relatime)
cgroup2 on /host-sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime)
```

cgroup1 are disabled (k8s admins confirmed).

Kubernetes version:
```
$ kubectl version --output=yaml
...
serverVersion:
  buildDate: "2024-03-15T00:00:02Z"
  compiler: gc
  gitCommit: fc11ff34c34bc1e6ae6981dc1c7b3faa20b1ac2d
  gitTreeState: clean
  gitVersion: v1.28.8
  goVersion: go1.21.8
  major: "1"
  minor: "28"
  platform: linux/amd64
```

OS version:
```
$ cat /etc/issue
Ubuntu 22.04.4 LTS
```

Is this information enough to confirm or debug?
If you need to change the parameters and try it, tell me, I’ll do it.
